fontforge/fvfonts.c

```diff
@@ -1079,62 +1079,60 @@ static void FVMergeRefigureMapSel(FontViewBase *fv,SplineFont *into,SplineFont *
 
 static void _MergeFont(SplineFont *into,SplineFont *other,struct sfmergecontext *mc) {
     int i, cnt, doit, emptypos, index, k;
-    SplineFont *o_sf, *bitmap_into;
+    SplineFont *bitmap_into;
     BDFFont *bdf;
     FontViewBase *fvs;
     int *mapping;
 
     emptypos = into->glyphcnt;
+
     mapping = malloc(other->glyphcnt*sizeof(int));
-    memset(mapping,-1,other->glyphcnt*sizeof(int));
+    for (i = 0; i<other->glyphcnt; ++i)
+	mapping[i] = -1;
 
     bitmap_into = into->cidmaster!=NULL? into->cidmaster : into;
 
+    /* If other starts as CID it should be flattened before calling */
+    cnt = 0;
     for ( doit=0; doit<2; ++doit ) {
```

Comment:
Everything below to (green) line 1135 is the same except that the `do ... while()` loop level is removed. The code used to "loop once" with `other` if there were no subfonts and over the subfonts if there were. The new code just uses other as the font. `o_sf` is accordingly eliminated and that string replaced with `other`.